packages/modular-scripts/src/build/buildPackage/makeBundle.ts

```diff
@@ -183,7 +183,10 @@ export async function makeBundle(
           // Let's collect the name and add it in the package.json
           // we publish to the registry
           // TODO: make sure local workspaces are NOT explicitly included in package.json
-          if (packageJsons[importedPackage].private !== true) {
+          if (
+            includePrivate ||
+            packageJsons[importedPackage].private !== true
+          ) {
             localImports[importedPackage] = packageJsons[importedPackage]
```

Comment:
Maybe let's add a console warning here if we are in the case where we includePrivate and also hit this branch?

Reply:
I think it's perfectly reasonable to break projects down into multiple packages that it wouldn't make sense to publish on their own, so I don't think a warning is warranted every time someone builds a project that relies on unpublished packages in this way.

Reply:
The use case of building a package for publishing implies that all the dependencies are being published and maybe worth considering if a packages should be public or not. I wouldn’t expect anyone to be using the private flag for building on packages which aren’t being published. 

Given that private package building is opt in I would put a warning to show that the package being built is depending on other pieces which would need to be build opt in.

Reply:
I think there's a distinction between publishing the source and building a distribution. The private field is simply a way of specifying that a package should not be published standalone and implies nothing about the build process or how it's otherwise used.

I presume users will most commonly add the private flag after encountering a build failure where they used private package dependencies in their project.

Given the above, it still strikes me as odd to repeatedly display a warning for something the user has opted into.

Reply:
I'd suggest we merge as-is and split out a new feature of debug logging that would be useful for notice kind of messages like this.